src/popup/pages/Names/Auction.vue

```diff
@@ -35,44 +43,51 @@ export default {
   props: {
     name: { type: String, required: true },
   },
-  data: () => ({ loading: true }),
-  mounted() {
-    const id = setInterval(() => this.updateAuctionEntry(), 3000);
-    this.$once('hook:destroyed', () => clearInterval(id));
-    this.$watch(
-      ({ name }) => name,
-      () => {
-        this.loading = true;
-        this.updateAuctionEntry();
-      },
-      { immediate: true },
-    );
-  },
-  methods: {
-    async updateAuctionEntry() {
-      await watchUntilTruthy(() => this.$store.state.middleware);
+  setup(props, { root }) {
+    const { getMiddleware } = useMiddleware({ store: root.$store });
+
+    const loading = ref(true);
+
+    async function updateAuctionEntry() {
+      loading.value = true;
+      const middleware = await getMiddleware();
       try {
-        const res = await this.$store.state.middleware.getNameById(this.name);
+        const res = await middleware.getNameById(props.name);
         const { auctionEnd, bids } = res.info;
-        const loadedBids = await Promise.all(bids.map(async (txId) => {
-          const { tx } = await this.$store.state.middleware.getTxByIndex(txId);
+        const loadedBids = await Promise.all(bids.map(async (txId: any) => {
```

Comment:
If it's `txId` aren't we able to retrieve it's type (number)?

Reply:
Sure, makes sense.